src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -131,43 +151,46 @@ static void rtc_irq_handler(void) {
     }
 }
 
-static bool rtc_alarm_repeats(datetime_t *t) {
-    // If any value is set to -1 then we don't match on that value
-    // hence the alarm will eventually repeat
-    if (t->year  < 0) return true;
-    if (t->month < 0) return true;
-    if (t->day   < 0) return true;
-    if (t->dotw  < 0) return true;
-    if (t->hour  < 0) return true;
-    if (t->min   < 0) return true;
-    if (t->sec   < 0) return true;
-    return false;
+// return != 0 on repeat, -1 on repeat every second
+static int8_t rtc_alarm_repeats(const datetime_t *t) {
+  // If any value is set to -1 then we don't match on that value
+  // hence the alarm will eventually repeat
+    if((t->year & t->month & t->day & t->dotw & t->hour & t->min & t->sec) < 0) return CONTINUOUS_REPEAT_ON_SEC;
```

Comment:
Graham might feel differently to me, but I wonder if explicitly doing
```
if (t->year < 0 && t->month < 0 && t->day < 0 && t->dotw < 0 && t->hour < 0 && t->min < 0 && t->sec < 0)
```
might be more readable than using bitwise-AND ?
(your version is also missing the space between the `if` and the `(` :slightly_smiling_face: )

Reply:
Right, I already changed it, just not pushed

Reply:
> it needs nearly twice the operations and therefore cycles to check

But with RP2040 running at 125MHz, and the RTC alarm only firing at most once per second, I don't think that should be a huge problem? :wink:

Reply:
I am simply no friend of increasing the number of execution cycles just to increase readability. If it's not clear, it could be documented.

Reply:
Another Graham-decision I think :wink:

Reply:
It could, I would be happy to change it if accepted. But of course it's a little bit bit "hacking" at a place where it is not really time critical. Just a few clock cycles and a few bytes binary code size reduction.